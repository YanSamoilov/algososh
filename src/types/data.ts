import { ElementStates } from "./element-states";

export interface symbolProps {
    symbol: string;
    state: ElementStates;
}

export interface numsProps {
    num: number;
    state: ElementStates;
}

export interface stringCharsProps {
  adding?: boolean;
  deleting?: boolean;
  noArrow?: boolean;
  tail?: string;
  head?: string;
  char?: string;
  state: ElementStates;
  extraCircle?: {
    char: string;
  }
}

export type pagesArrayProps = symbolProps | numsProps;
