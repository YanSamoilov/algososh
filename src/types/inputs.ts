
export enum InputsByListPage {
  Value = "value",
  Index = "index"
}
