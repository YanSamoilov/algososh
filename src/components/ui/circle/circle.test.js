import { Circle } from "./circle";
import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";

describe("Тестирование компонента Circle", () => {
  describe("Рендер с буквами и без букв", () => {
    it("Circle без буквы рендерится без ошибок", () => {
      const tree = renderer.create(<Circle />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("Circle c буквами рендерится без ошибок", () => {
      const tree = renderer.create(<Circle letter="Symbols" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("Circle рендериться с head", () => {
    it("Circle cо строкой в head рендерится без ошибок", () => {
      const tree = renderer.create(<Circle head={"head"} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Circle c node в head рендерится без ошибок", () => {
      const tree = renderer.create(<Circle head={<Circle />} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("Circle рендериться с tail", () => {
    it("Circle cо строкой в tail рендерится без ошибок", () => {
      const tree = renderer.create(<Circle tail={"test"} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Circle c node в tail рендерится без ошибок", () => {
      const tree = renderer.create(<Circle tail={<Circle />} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("Circle рендериться в разных состояниях", () => {
    it("Circle в состоянии default рендерится без ошибок", () => {
      const tree = renderer
        .create(<Circle state={ElementStates.Default} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Circle в состоянии Changing рендерится без ошибок", () => {
      const tree = renderer
        .create(<Circle state={ElementStates.Changing} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Circle в состоянии Modified рендерится без ошибок", () => {
      const tree = renderer
        .create(<Circle state={ElementStates.Modified} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Circle в состоянии Default рендерится без ошибок", () => {
      const tree = renderer
        .create(<Circle state={ElementStates.Default} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Circle в состоянии Active рендерится без ошибок", () => {
      const tree = renderer
        .create(<Circle state={ElementStates.Active} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  it("Circle с индексом рендерится без ошибок", () => {
    const tree = renderer.create(<Circle index={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Circle с пропом isSmall рендерится без ошибок", () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
