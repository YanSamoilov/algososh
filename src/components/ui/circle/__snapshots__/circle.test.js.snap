// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle Circle рендериться в разных состояниях Circle в состоянии Active рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   active"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle рендериться в разных состояниях Circle в состоянии Changing рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle рендериться в разных состояниях Circle в состоянии Default рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle рендериться в разных состояниях Circle в состоянии Modified рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle рендериться в разных состояниях Circle в состоянии default рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle рендериться с head Circle c node в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle рендериться с head Circle cо строкой в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle рендериться с tail Circle c node в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle Circle рендериться с tail Circle cо строкой в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test
  </div>
</div>
`;

exports[`Тестирование компонента Circle Circle с индексом рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Circle с пропом isSmall рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер с буквами и без букв Circle c буквами рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Symbols
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Рендер с буквами и без букв Circle без буквы рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
